pictorus-blocks/src/std_blocks/fmu_block.rs

```diff
@@ -138,117 +139,31 @@ impl From<fmu_runner::FmuUnpackError> for FmuErrors {
-///         self.data = output.iter().map(|&x| OldBlockData::from_scalar(x)).collect();
-///         output[0]
-///     }
-/// }
-/// ```
-macro_rules! impl_process_block {
-    ($n_in:tt, $n_out:tt) => {
-        impl ProcessBlock for FmuBlock<$n_in, $n_out> {
-            type Parameters = Parameters;
-            type Inputs = impl_process_block!(@tuple_type, f64, $n_in);
-            type Output = impl_process_block!(@tuple_type, f64, $n_out);
-            #[allow(unused_variables)]
-            fn process<'b>(
-                &'b mut self,
-                parameters: &Self::Parameters,
-                context: &dyn Context,
-                inputs: pictorus_traits::PassBy<'_, Self::Inputs>,
-            ) -> pictorus_traits::PassBy<'b, Self::Output> {
-                let output: [f64; $n_out] = self.run_time_step(
-                    parameters,
-                    context,
-                    impl_process_block!(@input_spec, $n_in, inputs),
-                );
-                self.data.clear();
-                self.data = output
-                    .iter()
-                    .map(|&x| OldBlockData::from_scalar(x))
-                    .collect();
-                impl_process_block!(@output_spec, $n_out, output)
-            }
-        }
-    };
-
-    // Special casing here for passing length 0 and length 1 inputs to `run_time_step`
-    (@input_spec, 0, $name:expr) => {&[]};
-    (@input_spec, 1, $name:expr) => {&[$name]};
-    (@input_spec, $n:expr, $name:expr) => {&$name.try_into().expect("This is a known size")};
-
-    // Special casing here for returning length 0 and length 1 outputs
-    (@output_spec, 0, $name:expr) => {()};
-    (@output_spec, 1, $name:expr) => {$name[0]};
-    (@output_spec, $n:expr, $name:expr) => {
-        $name.try_into().expect("This is a known size")
-    };
-
-
-    // These just allow one to refer to the tuple type by its length
-    (@tuple_type, $type:ty, 0) => {()};
-    (@tuple_type, $type:ty, 1) => {$type};
-    (@tuple_type, $type:ty, 2) => {($type, $type)};
-    (@tuple_type, $type:ty, 3) => {($type, $type, $type)};
-    (@tuple_type, $type:ty, 4) => {($type, $type, $type, $type)};
-    (@tuple_type, $type:ty, 5) => {($type, $type, $type, $type, $type)};
-    (@tuple_type, $type:ty, 6) => {($type, $type, $type, $type, $type, $type)};
-    (@tuple_type, $type:ty, 7) => {($type, $type, $type, $type, $type, $type, $type)};
-    (@tuple_type, $type:ty, 8) => {($type, $type, $type, $type, $type, $type, $type, $type)};
+impl<const N_IN: usize, const N_OUT: usize> ProcessBlock for FmuBlock<N_IN, N_OUT> {
+    type Parameters = Parameters;
+    // We use homogeneous arrays for inputs and outputs to avoid the limits/complexity
+    // of mixed data types. This is safe to do because we only support FMI 2.0 which
+    // only supports scalar values. If we add support for FMI 3.0 we will need to revisit this.
+    type Inputs = [f64; N_IN];
+    type Output = [f64; N_OUT];
```

Comment:
For some reason I had really though we didn't have [f64;N] impl Pass, but looks like we did and it is work well here